apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,77 @@
+import { IPreferenceChannels, StepTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepResponseDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      tags: template.tags,
+      active: template.active,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      steps: template.steps.map(WorkflowTemplateGetMapper.toStepResponseDto),
+      name: template.name,
+      description: template.description,
+      origin: template.origin || WorkflowOriginEnum.NOVU,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+    };
+  }
+  static toMinifiedWorkflowDto(template: NotificationTemplateEntity): MinifiedResponseWorkflowDto {
+    return {
+      _id: template._id,
+      name: template.name,
+      tags: template.tags,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+      stepSummery: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
+    };
+  }
+
+  static toWorkflowsMinifiedDtos(templates: NotificationTemplateEntity[]): MinifiedResponseWorkflowDto[] {
+    return templates.map(WorkflowTemplateGetMapper.toMinifiedWorkflowDto);
+  }
+  static toStepResponseDto(step: NotificationStepEntity): StepResponseDto {
+    return {
+      name: step.name || 'Missing Name',
+      stepUuid: step._templateId,
+      type: step.template?.type || StepTypeEnum.EMAIL,
+      controls: WorkflowTemplateGetMapper.convertControls(step),
+      active: step.active || true,
+      shouldStopOnFail: step.shouldStopOnFail || true,
+      controlValues: step.controlVariables || {},
+    };
+  }
+
+  private static convertControls(step: NotificationStepEntity) {
+    if (step.template?.controls) {
+      return { schema: step.template.controls.schema };
+    } else {
+      return undefined;
+    }
+  }
+
+  private static toPreferences(preferenceSettings: IPreferenceChannels): DiscoverWorkflowOutputPreferences {
+    const { email } = preferenceSettings;
+    const { sms } = preferenceSettings;
+    const { push } = preferenceSettings;
+    const { chat } = preferenceSettings;
+    const inApp = preferenceSettings.in_app;
```

Comment:
move all to one line

Reply:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** This is a bit untidy, it's unclear why we are mapping between 2 different cases needlessly right now. Can we declare the usage of `preferenceSettings` in line on the returned object?‏

Reply:
it has been removed completly